File: src/main/java/pw/react/backend/services/UserService.java
package pw.react.backend.services;

import pw.react.backend.models.UserEntity;

import java.util.Collection;

public interface UserService {
    UserEntity validateAndSave(UserEntity user);
    UserEntity updatePassword(UserEntity user, String password);
    Collection<UserEntity> batchSave(Collection<UserEntity> users);
}

File: src/main/java/pw/react/backend/services/UserService.kt
package pw.react.backend.services

import pw.react.backend.models.domain.User


interface UserService {
    fun findById(id: Long): User?
    fun findUserByEmail(email: String): User?
    fun validateAndSave(user: User): User
    fun updateName(user: User, name: String): User
    fun updateLastName(user: User, lastName: String): User
    fun updatePassword(user: User, password: String): User
    fun saveUnique(user: User): User
    fun batchSave(users: List<User>): List<User>
}
